File: auction-app-backend/src/main/java/com/internship/auctionapp/services/category/CategoryService.java
package com.internship.auctionapp.services.category;

import com.internship.auctionapp.models.Category;
import com.internship.auctionapp.requests.CreateCategoryRequest;

import java.util.List;
import java.util.UUID;

public interface CategoryService {
    Category addCategory(CreateCategoryRequest createCategoryRequest);

    List<Category> getAllCategories();

    void deleteCategory(UUID categoryId);
}

File: auction-app-ui/src/services/categoryService.ts
import agent from 'lib/agent';
import { Category } from 'models/category';
import { CreateCategoryRequest } from 'models/request/create/createCategoryRequest';

const BASE_URL = '/categories';

const categoryService = {
  getCategories: () => agent.get<Category[]>(BASE_URL),
  addCategory: (createCategoryRequest: CreateCategoryRequest) => agent.post<any>(BASE_URL, createCategoryRequest),
  deleteCategory: (categoryId: string) => agent.delete<any>(`${BASE_URL}/${categoryId}`),
};

export default categoryService;
